libcodechecker/analyze/store_handler.py

```diff
@@ -343,44 +343,76 @@ def addReport(storage_session,
+                    report.file_id = file_id
+
+                elif report.detection_status == 'unresolved':
+                    report.detection_status = 'unresolved'
+                    report.file_id = file_id
 
-                if report.detection_status == 'new' and not \
-                        storage_session.is_touched(run_id, report.id) or \
-                        report.detection_status == 'unresolved' or \
-                        report.detection_status == 'reopened':
-                    new_status = 'unresolved'
+                elif report.detection_status == 'reopened':
+                    report.detection_status = 'unresolved'
                     report.file_id = file_id
-                    change_path_and_events(session,
-                                           report.id,
-                                           bugpath,
-                                           events)
+
                 elif report.detection_status == 'resolved':
-                    new_status = 'reopened'
+                    report.detection_status = 'reopened'
                     report.file_id = file_id
                     report.fixed_at = None
-                    change_path_and_events(session,
-                                           report.id,
-                                           bugpath,
-                                           events)
-
-                if new_status:
-                    report.detection_status = new_status
-
-                storage_session.touch_report(run_id, report.id)
-
-                return report.id
-
-        LOG.debug("no duplicate storing report")
+            else:
+                # There is already a set status for this hash in this
+                # storage session, use it for the reports with the same
+                # hash.
+                report.detection_status = detection_status_for_hash
+
+            # Update the stored path in the database
+            # if the same bug path was found.
+            if report.checker_id == checker_id and \
+                    is_same_event_path(report.id, events, session):
+                change_path_and_events(session,
+                                       report.id,
+                                       bugpath,
+                                       events)
+
+                # The bug hash and the path is the same, this is a
+                # duplicate.
+                duplicate_report_id = report.id
+            else:
+                # Only hash is the same but not a path duplicate
+                # report was found, will be stored as a new report
+                # but the already available detection status will be used.
+                new_report_status = report.detection_status
+
+            # We found the hash during this storage session, mark it
+            # so the detection status won't be set to resolved
+            # at the end of the storage session.
+            storage_session.save_detection_status(run_id,
```

Comment:
Do we need to save the detection status if we already set a detection status? See line 385.